packages/tina-graphql-gateway/src/auth/TinaCloudProvider.tsx

```diff
@@ -105,12 +107,24 @@ export const AuthWallInner = ({
  *
  * Note: this will not restrict access for local filesystem clients
  */
-export const TinaCloudAuthWall = (props: TinaCloudAuthWallProps) => {
+export const TinaCloudAuthWall = (
+  props: TinaCloudAuthWallProps & CreateClientProps
+) => {
   useTinaAuthRedirect()
+  const cms =
```

Comment:
Related to [this comment](https://github.com/tinacms/tina-graphql-gateway/pull/254/files#r651045524)
What if we instead did something like:

```
let cms = props.cms || new TinaCMS({  enabled: true, sidebar: true })
cms.apis.tina = createClient(props)
```

This way, even if you needed to pass a custom CMS object into `TinaCloudAuthWall`, the client is always created consistently in the same way (through the `TinaCloudAuthWallProps` ). If we were worried about users wanting to attach the tina api onto the CMS object, we could handle that with:

```
let cms = props.cms || new TinaCMS({  enabled: true, sidebar: true })
if(!cms.apis.tina) {
    cms.apis.tina = createClient(props)
}
```

Reply:
@jamespohalloran yup 🎉 this makes perfect sense to me!